bob-backend/src/pearl/group.rs

```diff
@@ -474,14 +474,36 @@ impl Group {
         String::from_utf8(hex).unwrap()
     }
 
+    pub(crate) async fn find_oldest_inactive_holder(&self) -> Option<Holder> {
+        let holders_lock = self.holders();
+        let holders = holders_lock.read().await;
+        let mut result: Option<Holder> = None;
+        let period = self.settings.timestamp_period_as_secs() / 2;
```

Comment:
I suggest to add 10 seconds to this value to better handle cases, when timestamp period is too small